### PR TITLE
Added GermEval 18 Offensive Language dataset

### PR DESCRIPTION
**Pull request by lukasgarbas**
Added GermEval 2018 data for binary text classification of German tweets. The dataset was taken from https://github.com/uds-lsv/GermEval-2018-Data.

### CODE REVIEW

File: flair/datasets/document_classification.py [hunk 1]
```diff
@@ -1541,6 +1541,73 @@ def __init__(self,
...
+class GERMEVAL_2018_OFFENSIVE_LANGUAGE(ClassificationCorpus):
+    """
+    GermEval 2018 Task 1 corpus for identification of offensive language.
+    Classifying German tweets into 2 coarse-grained categories OFFENSIVE or OTHER.
+    """
+
+    def __init__(self,
+                 base_path: Union[str, Path] = None,
+                 memory_mode: str = 'full',
+                 tokenizer: Union[bool, Callable[[str], List[Token]], Tokenizer] = SpaceTokenizer(),
```
Comment (alanakbik) [inline flair/datasets/document_classification.py:1553]: Perhaps change the default tokenizer to `SegtokTokenizer`? It seems the tweets are not tokenized, so the whitespace tokenizer will create some words that include punctuation.

File: flair/datasets/document_classification.py [hunk 2]
```diff
@@ -1541,6 +1541,73 @@ def __init__(self,
         )
 
 
+class GERMEVAL_2018_OFFENSIVE_LANGUAGE(ClassificationCorpus):
+    """
+    GermEval 2018 Task 1 corpus for identification of offensive language.
+    Classifying German tweets into 2 coarse-grained categories OFFENSIVE or OTHER.
+    """
+
+    def __init__(self,
```
Comment (alanakbik) [inline flair/datasets/document_classification.py:1550]: Could you add one more boolean constructor argument called `fine_grained_classes` (or so) that's set to `False` by default? If set to `True` you could load the fine-grained offense classes instead. 

To do this in the current logic, you'd need to generate separate FastText-formatted files for the fine-grained case.

**Review by alanakbik — CHANGES_REQUESTED** [review]
Thanks for adding this! Could you check out the review comments?

File: flair/datasets/document_classification.py (hunk 1 above)
Comment (lukasgarbas) [reply flair/datasets/document_classification.py]: Done! Having SegtokTokenizer as the default makes more sense for tweets.

File: flair/datasets/document_classification.py (hunk 2 above)
Comment (lukasgarbas) [reply flair/datasets/document_classification.py:1551]: Sure. I added support for the fined-grained case. This way one could use the data from both GermEval 2018 tasks (binary and 4-way classification). As you've mentioned before, this requires creating separate files for both tasks.

**Review by alanakbik — APPROVED** [review]

**Comment by alanakbik** [comment]
@lukasgarbas thanks for adding this!

**Pull request merged**